tokio/src/runtime/blocking/pool.rs

```diff
@@ -101,19 +101,27 @@ impl BlockingPool {
     pub(crate) fn spawner(&self) -> &Spawner {
         &self.spawner
     }
-}
 
-impl Drop for BlockingPool {
-    fn drop(&mut self) {
+    pub(crate) fn shutdown(&mut self, timeout: Option<Duration>) {
         let mut shared = self.spawner.inner.shared.lock().unwrap();
 
+        if shared.shutdown {
```

Comment:
The function can be called multiple times. First, by explicitly calling `shutdown` then by the drop handler calling `shutdown`. This prevents shutting down twice.

Reply:
Might be good if there was a comment in the code stating that, as well as on the PR?